sympl/_core/util.py

```diff
@@ -639,6 +638,141 @@ def __getattr__(self, item):
         return getattr(self._prognostic, item)
 
 
+class ScaledInputOutputWrapper(object):
+    """
+    Wraps any component and scales either inputs, outputs or tendencies
+    by a floating point value.
+
+    Example
+    -------
+
+    This is how the ScaledInputOutputWrapper can be used to wrap a Prognostic.
+
+    >>> scaled_component = ScaledInputOutputWrapper(
+    >>>     RRTMRadiation(),
+    >>>     input_scale_factors = {
+    >>>         'specific_humidity' = 0.2},
+    >>>     tendency_scale_factors = {
+    >>>         'air_temperature' = 1.5})
+    """
+
+    def __init__(self,
+                 component,
+                 input_scale_factors=None,
+                 output_scale_factors=None,
+                 tendency_scale_factors=None):
```

Comment:
Should also take in diagnostic_scale_factors and scale diagnostics. There's a distinction between diagnostics (from the current timestep) and outputs (from the next timestep).